cranelift-filetests/src/function_runner.rs

```diff
@@ -1,53 +1,301 @@
+            types::I8 => {
+                let data = p.match_constant_data(ty).unwrap();
+                Self::I8(ConstantDataU8::try_into(data).unwrap())
+            }
+            types::I16 => {
+                let data = p.match_constant_data(ty).unwrap();
+                Self::I16(ConstantDataU16::try_into(data).unwrap())
+            }
+            types::I32 => {
+                let data = p.match_constant_data(ty).unwrap();
+                Self::I32(ConstantDataU32::try_into(data).unwrap())
+            }
+            types::I64 => {
+                let data = p.match_constant_data(ty).unwrap();
+                Self::I64(ConstantDataU64::try_into(data).unwrap())
+            }
+            types::I128 => {
+                let data = p.match_constant_data(ty).unwrap();
+                let data = ConstantDataU128::try_into(data).unwrap();
+                Self::I128(data.rotate_left(64))
+            }
+            types::F32 => {
+                let data = consume!(ty, p.match_ieee32("Expected a 32-bit float").unwrap());
+                let data = ConstantDataU32::try_into(data).unwrap();
+                Self::F32(f32::from_bits(data))
+            }
+            types::F64 => {
+                let data = consume!(ty, p.match_ieee64("Expected a 64-bit float").unwrap());
+                let data = ConstantDataU64::try_into(data).unwrap();
+                Self::F64(f64::from_bits(data))
+            }
+            b if b.is_bool() => {
+                let data = consume!(
+                    ty,
+                    boolean_to_vec(p.match_bool("Expected a boolean").unwrap())
+                );
+                let data = ConstantDataU8::try_into(data).unwrap();
+                Self::Bool(data != 0)
+            }
+            _ => {
+                return Err(format!(
+                    "Expected a type of: float, int or bool, found {}.",
+                    ty
+                ))
+            }
+        };
+        Ok(constant_data)
+    }
 }
 
 impl FunctionRunner {
-    /// Build a function runner from a function and the ISA to run on (must be the host machine's ISA)
-    pub fn new(function: Function, isa: Box<dyn TargetIsa>) -> Self {
-        Self { function, isa }
+    /// Build a function runner from an action, a function and the ISA to run on (must be the host machine's ISA).
+    pub fn new(function: Function, isa: Box<dyn TargetIsa>, action: FunctionRunnerAction) -> Self {
+        Self {
+            function,
+            isa,
+            action,
+        }
     }
 
-    /// Build a function runner using the host machine's ISA and the passed flags
-    pub fn with_host_isa(function: Function, flags: settings::Flags) -> Self {
-        let builder = host_isa_builder().expect("Unable to build a TargetIsa for the current host");
-        let isa = builder.finish(flags);
-        Self::new(function, isa)
+    fn invoke<T>(code_page: &Mmap) -> T
+    where
+        T: std::fmt::Display,
```

Comment:
It this requirement actually needed here?

Reply:
No, remnant from debugging, will remove.